Add top level protocol packet description.

iphydf opened the pull request:
All other packets sit inside this packet.
Yes, there will be redundancy between this part of the spec and the
detailed sections on each packet format, as both will define the packet
kind number. I think this is acceptable. Maybe later we'll decide that
it's not, but for now the table seems useful for implementations
(perhaps we can generate it later).

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/toktok/tox-spec/19)
<!-- Reviewable:end -->

zetok commented:
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions.

---

*[spec.md, line 204 \[r1\]](https://reviewable.io:443/reviews/toktok/tox-spec/19#-KBiFKZa9wKw5x_Hx_IS:-KBiFKZa9wKw5x_Hx_IT:-132086128) ([raw file](https://github.com/toktok/tox-spec/blob/5f58d90bc95fafef27e7708cf025ce920283b9c3/spec.md#L204)):*
Shouldn't that have a different name, e.g. `Packet Type` or `Kind` ?

"Protocol" packet makes it sound as if section was supposed to be about packet that points at transport protocol (UDP/TCP), or at least that's how I perceive it.

Especially given the preceding part:
> ```
> ## Protocol
> A Protocol is a transport layer protocol directly below the Tox protocol itself.
> ```

Kinda confusing.

---

*[spec.md, line 235 \[r1\]](https://reviewable.io:443/reviews/toktok/tox-spec/19#-KBiNwvOunhn3IOZowjB:-KBiNwvPQTs9aGMWkVtS:-2090979037) ([raw file](https://github.com/toktok/tox-spec/blob/5f58d90bc95fafef27e7708cf025ce920283b9c3/spec.md#L235)):*
I think that there are some things missing, e.g. 
```
### Onion packet (0x08)

Same format as initial onion packet but packet id is 0x08 instead of 0x80.

### Onion packet response (0x09)

Same format as onion packet but packet id is 0x09 instead of 0x8e.

### Data (0x10 and up)
```

Given that list is supposed to be exhaustive, they should be included..?

---

*[spec.md, line 239 \[r1\]](https://reviewable.io:443/reviews/toktok/tox-spec/19#-KBiLf87glc-m6kiPcJA:-KBiLf87glc-m6kiPcJB:620060803) ([raw file](https://github.com/toktok/tox-spec/blob/5f58d90bc95fafef27e7708cf025ce920283b9c3/spec.md#L239)):*
>The packet id for encrypted packets is 27.

It would seem like `0x1b` is a "general-purpose" crypto packet? Isn't that just a `Crypto` packet?

---

*[spec.md, line 240 \[r1\]](https://reviewable.io:443/reviews/toktok/tox-spec/19#-KBiKJDv48ZhUlzo0_fl:-KBiKJDv48ZhUlzo0_fm:1295496523) ([raw file](https://github.com/toktok/tox-spec/blob/5f58d90bc95fafef27e7708cf025ce920283b9c3/spec.md#L240)):*
From where that comes from?

I.e. there's `DHT Request` packet that starts with `0x20`, and there's `Friend request packet when sent as an onion data packet` that's also `0x20`.
By quickly skimming though some text I wasn't able to find anything other that would be relevant.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/toktok/tox-spec/19)*
<!-- Sent from Reviewable.io -->

iphydf commented:
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions.

---

*[spec.md, line 204 \[r1\]](https://reviewable.io:443/reviews/toktok/tox-spec/19#-KBiFKZa9wKw5x_Hx_IS:-KBiZ9I8cKvafqvYhemq:593266148) ([raw file](https://github.com/toktok/tox-spec/blob/5f58d90bc95fafef27e7708cf025ce920283b9c3/spec.md#L204)):*
Rewrote the section a bit and changed Protocol in the previous section to Transport Protocol as disambiguation.

---

*[spec.md, line 235 \[r1\]](https://reviewable.io:443/reviews/toktok/tox-spec/19#-KBiNwvOunhn3IOZowjB:-KBiZLg3w6aSDr6liJFX:76022575) ([raw file](https://github.com/toktok/tox-spec/blob/5f58d90bc95fafef27e7708cf025ce920283b9c3/spec.md#L235)):*
Those are all inside other packets, not in the top level Protocol Packet. The packet kinds you pointed at are inside TCP client packets. There are a number of other packet types with their own kinds (messenger has some, group chat has yet another nested packet format). This list is exhaustive for the top level packet kinds.

---

*[spec.md, line 239 \[r1\]](https://reviewable.io:443/reviews/toktok/tox-spec/19#-KBiLf87glc-m6kiPcJA:-KBi_VlqOObiVqINKcZ4:1992371134) ([raw file](https://github.com/toktok/tox-spec/blob/5f58d90bc95fafef27e7708cf025ce920283b9c3/spec.md#L239)):*
True, the Crypto packet in this list is the DHT Request packet. I'll change that. I do think Crypto Data is a good name, though.

---

*[spec.md, line 240 \[r1\]](https://reviewable.io:443/reviews/toktok/tox-spec/19#-KBiKJDv48ZhUlzo0_fl:-KBi_iiggn8dLSu7PbSS:-499146009) ([raw file](https://github.com/toktok/tox-spec/blob/5f58d90bc95fafef27e7708cf025ce920283b9c3/spec.md#L240)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/toktok/tox-spec/19)*
<!-- Sent from Reviewable.io -->

zetok commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/toktok/tox-spec/19)*
<!-- Sent from Reviewable.io -->
